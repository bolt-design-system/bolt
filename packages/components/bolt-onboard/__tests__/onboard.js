import { render, stopServer, renderWC } from '../../../testing/testing-helpers';
import schema from '../onboard.schema';
const componentSelector = 'bolt-onboard';
const { status, size, radius } = schema.properties;
let page, fixtures;

afterAll(async () => {
  await stopServer();
  await page.close();
}, 100);

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const defaultData = {
    content: 'This is an onboard component',
  };

  fixtures = {
    defaultData,
  };
});

describe('Bolt onboard', () => {
  // With a Web Component
  test(`default`, async () => {
    const results = await render('@bolt-components-onboard/onboard.twig', {
      ...fixtures.defaultData,
    });

    const { innerHTML, outerHTML } = await renderWC(
      componentSelector,
      results.html,
      page,
    );

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
    await expect(innerHTML).toMatchSnapshot();
    await expect(outerHTML).toMatchSnapshot();
  });

  // Without a Web Component
  test(`default`, async () => {
    const results = await render('@bolt-components-onboard/onboard.twig', {
      ...fixtures.defaultData,
    });

    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});

describe('Bolt onboard prop -', () => {
  // Target each of the schema keys with the following pattern
  status.enum.forEach(async option => {
    test(`status: ${option}`, async () => {
      const results = await render('@bolt-components-onboard/onboard.twig', {
        ...fixtures.defaultData,
        status: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });

  size.enum.forEach(async option => {
    test(`size: ${option}`, async () => {
      const results = await render('@bolt-components-onboard/onboard.twig', {
        ...fixtures.defaultData,
        size: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });

  radius.enum.forEach(async option => {
    test(`radius: ${option}`, async () => {
      const results = await render('@bolt-components-onboard/onboard.twig', {
        ...fixtures.defaultData,
        radius: option,
      });

      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
});
