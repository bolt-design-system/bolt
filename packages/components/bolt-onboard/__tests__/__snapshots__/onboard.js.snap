// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt onboard default 1`] = `
<bolt-onboard>
  <replace-with-children class="c-bolt-onboard c-bolt-onboard--status-information c-bolt-onboard--size-default c-bolt-onboard--radius-none">
    This is an onboard component
  </replace-with-children>
</bolt-onboard>
`;

exports[`Bolt onboard default 2`] = `
<div class="c-bolt-onboard c-bolt-onboard--status-information c-bolt-onboard--size-default c-bolt-onboard--radius-none">
  <slot>
  </slot>
</div>
`;

exports[`Bolt onboard default 3`] = `
<bolt-onboard>
  This is an onboard component
</bolt-onboard>
`;

exports[`Bolt onboard default 4`] = `
<bolt-onboard>
  <replace-with-children class="c-bolt-onboard c-bolt-onboard--status-information c-bolt-onboard--size-default c-bolt-onboard--radius-none">
    This is an onboard component
  </replace-with-children>
</bolt-onboard>
`;

exports[`Bolt onboard prop - radius: full 1`] = `
<bolt-onboard radius="full">
  <replace-with-children class="c-bolt-onboard c-bolt-onboard--status-information c-bolt-onboard--size-default c-bolt-onboard--radius-full">
    This is an onboard component
  </replace-with-children>
</bolt-onboard>
`;

exports[`Bolt onboard prop - radius: large 1`] = `
<bolt-onboard radius="large">
  <replace-with-children class="c-bolt-onboard c-bolt-onboard--status-information c-bolt-onboard--size-default c-bolt-onboard--radius-large">
    This is an onboard component
  </replace-with-children>
</bolt-onboard>
`;

exports[`Bolt onboard prop - radius: none 1`] = `
<bolt-onboard radius="none">
  <replace-with-children class="c-bolt-onboard c-bolt-onboard--status-information c-bolt-onboard--size-default c-bolt-onboard--radius-none">
    This is an onboard component
  </replace-with-children>
</bolt-onboard>
`;

exports[`Bolt onboard prop - size: default 1`] = `
<bolt-onboard size="default">
  <replace-with-children class="c-bolt-onboard c-bolt-onboard--status-information c-bolt-onboard--size-default c-bolt-onboard--radius-none">
    This is an onboard component
  </replace-with-children>
</bolt-onboard>
`;

exports[`Bolt onboard prop - size: large 1`] = `
<bolt-onboard size="large">
  <replace-with-children class="c-bolt-onboard c-bolt-onboard--status-information c-bolt-onboard--size-large c-bolt-onboard--radius-none">
    This is an onboard component
  </replace-with-children>
</bolt-onboard>
`;

exports[`Bolt onboard prop - size: medium 1`] = `
<bolt-onboard size="medium">
  <replace-with-children class="c-bolt-onboard c-bolt-onboard--status-information c-bolt-onboard--size-medium c-bolt-onboard--radius-none">
    This is an onboard component
  </replace-with-children>
</bolt-onboard>
`;

exports[`Bolt onboard prop - status: information 1`] = `
<bolt-onboard status="information">
  <replace-with-children class="c-bolt-onboard c-bolt-onboard--status-information c-bolt-onboard--size-default c-bolt-onboard--radius-none">
    This is an onboard component
  </replace-with-children>
</bolt-onboard>
`;

exports[`Bolt onboard prop - status: success 1`] = `
<bolt-onboard status="success">
  <replace-with-children class="c-bolt-onboard c-bolt-onboard--status-success c-bolt-onboard--size-default c-bolt-onboard--radius-none">
    This is an onboard component
  </replace-with-children>
</bolt-onboard>
`;

exports[`Bolt onboard prop - status: warning 1`] = `
<bolt-onboard status="warning">
  <replace-with-children class="c-bolt-onboard c-bolt-onboard--status-warning c-bolt-onboard--size-default c-bolt-onboard--radius-none">
    This is an onboard component
  </replace-with-children>
</bolt-onboard>
`;
